### PR TITLE
power implimentation

### PR DESCRIPTION
**Pull request by Keyshal-211**
## Description of changes
Please provide a short description of the change or if the change is sufficiently described in the GitHub issue, provide a link to the issue.

  
## How has this been tested?
Please describe the test strategy you followed.

- [ ] automated unit test
- [ ] automated integration test
- [ ] automated acceptance test
- [ ] manual, exploratory test

In case of manual test, please document the test well including a set of user instructions and prerequisites. Each including an action, it's result, and where appropriate a screenshot.
In case of automated tests, the test coverage should be reasonably high, especially on Service classes and other classes with complex logic.
    
## Checklist before requesting a review
- [X] My code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] My code fulfills all acceptance criteria
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] The test coverage (line and branch) is reasonably high, especially on Service classes and other classes with complex logic
- [ ] I have made corresponding changes to the documentation
- [ ] I have added explanation of architectural decision and rationales to [wiki/adr](https://github.com/IT-REX-Platform/wiki/tree/main/adr)
- [X] I have updated the changes in the ticket description

## Checklist for reviewer
- [ ] The code works and does not throw errors
- [ ] The code is easy to understand and there are no confusing parts
- [ ] The code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] The code change accomplishes what it is supposed to do
- [ ] I cannot think of any use case in which the code does not behave as intended
- [ ] The added and existing tests reasonably cover the code change
- [ ] I cannot think of any test cases, input or edge cases that should be tested in addition
- [ ] Description of the change is included in the documentation

### CODE REVIEW

File: src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/PowerScoreCalculator.java [hunk 1]
```diff
@@ -2,31 +2,66 @@
...
+        RewardScoreEntity health = allRewardScores.getHealth();
+        RewardScoreEntity fitness = allRewardScores.getFitness();
+        RewardScoreEntity power = allRewardScores.getPower();
+
+
+        double powerValueDouble = (growth.getValue() + strength.getValue()) + HEALTH_FITNESS_MULTIPLIER * (health.getValue() + fitness.getValue()) * (growth.getValue() + strength.getValue());
+        int powerValueInt = (int) Math.round(powerValueDouble);
+
+        RewardScoreLogEntry logEntry = RewardScoreLogEntry.builder()
+
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/PowerScoreCalculator.java:31]: the other fields of the log entry also have to be set! It is not allowed to have "null" in the database! For associated contents you can use an empty list.

File: src/main/resources/graphql/schema.graphqls [hunk 2]
```diff
@@ -117,4 +117,6 @@ enum RewardChangeReason {
     The associated contents are the content that are due for repetition.
     """
     CONTENT_DUE_FOR_REPETITION,
+    COMPOSITE_VALUE,
+    DONE_VALUE,
```
Comment (PaulBredl) [inline src/main/resources/graphql/schema.graphqls:121]: I would remove this and only use composite value. What does "DONE_VALUE" even mean? I think "COMPOSITE_VALUE" is fine for both calculations.

File: src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/PowerScoreCalculator.java [hunk 3]
```diff
@@ -2,31 +2,66 @@
...
+        power.getLog().add(logEntry);
+
+        return power;
 
-        return powerScoreBefore;
     }
 
     @Override
     public RewardScoreEntity calculateOnContentWorkedOn(AllRewardScoresEntity allRewardScores, List<Content> contents, UserProgressLogEvent event) {
-        RewardScoreEntity powerScoreBefore = allRewardScores.getPower();
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/PowerScoreCalculator.java:26]: the calculation is exactly the same as in the other method, this is huge code clone! Please extract that to a private method.

**Review by PaulBredl — CHANGES_REQUESTED** [review]

**Comment by PaulBredl** [comment]
Also you have to resolve merge conflicts

**Comment by PaulBredl** [comment]
The pull request description is just the template. If you don't fill out the template it would be better if you just remove it, otherwise it is confusing.

File: src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/PowerScoreCalculator.java (hunk 3 above)
Comment (Keyshal-211) [reply src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/PowerScoreCalculator.java:26]: done

File: src/main/resources/graphql/schema.graphqls (hunk 2 above)
Comment (Keyshal-211) [reply src/main/resources/graphql/schema.graphqls:121]: done

**Comment by Keyshal-211** [comment]
> Also you have to resolve merge conflicts

no I update it, I merged from main, no conflicts

**Review by PaulBredl — APPROVED** [review]

File: src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/PowerScoreCalculator.java [hunk 4]
```diff
@@ -2,31 +2,47 @@
...
+        RewardScoreEntity growth = allRewardScores.getGrowth();
+        RewardScoreEntity strength = allRewardScores.getStrength();
+        RewardScoreEntity health = allRewardScores.getHealth();
+        RewardScoreEntity fitness = allRewardScores.getFitness();
+        RewardScoreEntity power = allRewardScores.getPower();
+
+        double powerValueDouble = (growth.getValue() + strength.getValue()) + HEALTH_FITNESS_MULTIPLIER * (health.getValue() + fitness.getValue()) * (growth.getValue() + strength.getValue());
+        int powerValueInt = (int) Math.round(powerValueDouble);
+        RewardScoreLogEntry logEntry = RewardScoreLogEntry.builder()
+                .reason(RewardChangeReason.COMPOSITE_VALUE)
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/PowerScoreCalculator.java:40]: the other fields are still missing!

**Review by PaulBredl — CHANGES_REQUESTED** [review]
Fields in the log entry are missing

**Review by PaulBredl — APPROVED** [review]
looks good!
